lib/openscap_parser/group.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class Group < XmlNode
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title') &&
+        parsed_xml.at_css('title').text
+    end
+
+    def description
+      @description ||= @parsed_xml.at_css('description') &&
+        @parsed_xml.at_css('description').text
+    end
+
+    def requires
+      @requires ||= parsed_xml.at_css('requires') &&
+        parsed_xml.at_css('requires')["idref"].split
+    end
+
+    def conflicts
+      @conflicts ||= parsed_xml.at_css('conflicts') &&
+        parsed_xml.at_css('conflicts')["idref"]
+    end
+
+    def rationale
+      @rationale ||= newline_to_whitespace(
+        parsed_xml.at_css('rationale') &&
+          parsed_xml.at_css('rationale').text || ''
+      )
+    end
+
+    def weight
+      @weight ||= parsed_xml['weight']
+    end
+
+    def selected
+      @selected ||= parsed_xml['selected']
+    end
+
+    def parent_id
+      @parent_id = parsed_xml.xpath("name(..)='Group'") ? parsed_xml.xpath("../@id") : nil
+    end
```

Comment:
@skateman I left the `group_nodes` method the way that it was, going through every Group, and then I created this `parent_id` method in `group.rb` and `rule.rb` to check if the parent is a Group and if it is, set the parent_id to the parent Group's id. I did this because in the ancestry gem readme, it says that using the parent or parent_id is how you organize the tree structure. Does this look right?

Reply:
Nice catch :+1: I have no idea :see_no_evil: but I guess it's more efficient than an import with tree structure traversal recursion. Just remember these when something doesn't work so we can easily roll back to the recursion if needed :wink:

Reply:
Not sure if you considered the polymorphism, a parent of a group can be a group and the profile itself too, same goes for rules. As far as I remember how ancestry works, there should not be a problem on that side with it, we just have to cover it properly.

Here am saying hello to @kbrock and sendimg him my regards :wave: :heart: :smile:

Reply:
Yeah, that's true.. I'll look into ancestry a little more to make sure that polymorphism won't be an issue but I have a feeling it'll be able to handle it too, hopefully 😄

Reply:
If you use a hierarchical gem like ancestry, closure_trees, ... then `parent_id` is defined for you. Chances are you do not want to define `parent_id` with any of these. (and definitely not with `ancestry`)